src/main/java/com/javachallengers/simpson/service/SimpsonCharacterServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.javachallengers.simpson.service;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import com.javachallengers.simpson.exception.SimpsonCharacterException;
+import com.javachallengers.simpson.model.SimpsonCharacter;
+import com.javachallengers.simpson.model.dto.SimpsonCharacterRequestDTO;
+import com.javachallengers.simpson.repository.SimpsonCharacterRepository;
+
+@Service
+public class SimpsonCharacterServiceImpl implements SimpsonCharacterService {
+	private SimpsonCharacterRepository simpsonCharacterRepository;
+	
+	@Autowired
+	public SimpsonCharacterServiceImpl(SimpsonCharacterRepository simpsonCharacterRepository) {
+		super();
+		this.simpsonCharacterRepository = simpsonCharacterRepository;
+	}
+	
+	@Override
+	public Optional<SimpsonCharacter> getCharacterById(String id) {
+		Objects.requireNonNull(id, "id must not be null");
+		return simpsonCharacterRepository.findById(id);
+	}
+
+	@Override
+	public SimpsonCharacter createNewCharacter(SimpsonCharacterRequestDTO simpsonCharacterRequestDTO) throws SimpsonCharacterException {
+		Objects.requireNonNull(simpsonCharacterRequestDTO, "");
+		
+		String name = simpsonCharacterRequestDTO.getName();
```

Comment:
Can we use HibernateValidator to validate those fields?

Reply:
Yes we can improve to use bean validation.

Reply:
Great, thanks!

Reply:
Addded Bean validation in the DTO.